lib/components/ActionSheet/portal/index.tsx

```diff
@@ -27,38 +43,67 @@ const Portal = forwardRef<HTMLDivElement, PortalProps>(
     ) => {
         const { show, handleClose, className, position, type, expandable } =
             useContext(ActionSheetContext);
+
+        const [isVisible, setIsVisible] = useState(show);
+        const animationTimerRef = useRef<ReturnType<typeof setTimeout>>();
+
+        const toggleHandler = debounce(() => {
+            setIsVisible(!isVisible);
+            animationTimerRef.current = setTimeout(() => handleClose?.(), 500);
+        }, 200);
+
+        const childrenWithProps = React.Children.map(children, (child) => {
```

Comment:
I'm mapping through children. If it's a `Footer` (`ActionSheet.Footer`), then I'm cloning it and passing `handleClose` function from Parent (Portal ). Now, `Footer` won't use function from context, but get a new one  from prop. Other props won't be overwritten. 
Old function, which `Footer` was getting from context: `handleClose`.
New one, which `Footer` gets from props from the Parent:        
```
 const toggleHandler = debounce(() => {
            setIsVisible(!isVisible);
            animationTimerRef.current = setTimeout(() => handleClose?.(), 500);
        }, 200);
```